crates/muvm/src/guest/net.rs

```diff
@@ -33,64 +37,28 @@ pub fn configure_network() -> Result<()> {
-    };
-    if let Some(dhcpcd_path) = dhcpcd_path {
-        let output = Command::new(dhcpcd_path)
-            .args(["-M", "--nodev", "eth0"])
-            .output()
-            .context("Failed to execute `dhcpcd` as child process")?;
-        debug!(output:?; "dhcpcd output");
-        if !output.status.success() {
-            let err = if let Some(code) = output.status.code() {
-                anyhow!("`dhcpcd` process exited with status code: {code}")
-            } else {
-                anyhow!(
-                    "`dhcpcd` process terminated by signal: {}",
-                    output
-                        .status
-                        .signal()
-                        .expect("either one of status code or signal should be set")
-                )
-            };
-            Err(err)?;
-        }
-
-        return Ok(());
-    }
+    let address = Ipv4Addr::from_str(&netconf.address.context("Missing MUVM_NETWORK_ADDRESS")?)?;
+    let mask = u32::from(Ipv4Addr::from_str(
+        &netconf.mask.context("Missing MUVM_NETWORK_MASK")?,
+    )?);
+    let prefix = (!mask).leading_zeros() as u8;
+    let router = netconf.router.context("Missing MUVM_NETWORK_ROUTER")?;
+    let router = Ipv4Addr::from_str(&router)?;
 
-    let dhclient_path =
-        find_in_path("dhclient").context("Failed to check existence of `dhclient`")?;
-    let dhclient_path = if let Some(dhclient_path) = dhclient_path {
-        Some(dhclient_path)
-    } else {
-        find_executable("/sbin/dhclient")
-            .context("Failed to check existence of `/sbin/dhclient`")?
-    };
-    let dhclient_path =
-        dhclient_path.ok_or_else(|| anyhow!("could not find required `dhcpcd` or `dhclient`"))?;
-    let output = Command::new(dhclient_path)
-        .output()
-        .context("Failed to execute `dhclient` as child process")?;
-    debug!(output:?; "dhclient output");
-    if !output.status.success() {
-        let err = if let Some(code) = output.status.code() {
-            anyhow!("`dhclient` process exited with status code: {code}")
-        } else {
-            anyhow!(
-                "`dhclient` process terminated by signal: {}",
-                output
-                    .status
-                    .signal()
-                    .expect("either one of status code or signal should be set")
-            )
-        };
-        Err(err)?;
+    let (connection, handle, _) = new_connection().unwrap();
+    tokio::spawn(connection);
+    let mut links = handle.link().get().match_name("eth0".to_string()).execute();
+    if let Some(link) = links.try_next().await? {
+        handle
+            .address()
+            .add(link.header.index, IpAddr::V4(address), prefix)
+            .execute()
+            .await?;
+        handle.link().set(link.header.index).up().execute().await?
     }
+    handle.route().add().v4().gateway(router).execute().await?;
+    fs::write("/etc/resolv.conf", format!("nameserver {}", router))
```

Comment:
You are assuming that the default gateway and the dns server are the same. While true on most residential networks, it is not a correct assumption in general, and even there, some of us very intentionally override the ISP dns server to something else.

Reply:
This is inside the VM, so `passt` has already done its thing. In our current usage, `/etc/resolv.conf` inside the VM would contain a `nameserver` entry pointing to the host's default gateway...

https://man.archlinux.org/man/passt.1.en#Handling_of_traffic_with_local_destination_and_source_addresses

Reply:
> /etc/resolv.conf inside the VM would contain a nameserver entry pointing to the host's default gateway.

That is the problem that i am pointing out, in general nameserver != default gateway.

Reply:
More correct explanation here: https://github.com/AsahiLinux/muvm/issues/17#issuecomment-2160748250

Reply:
@WhatAmISupposedToPutHere Uhh, but I think I get what you mean now... The code here is only correct in the case of `passt` doing such DNS forwarding... If the resolver on the host is not a loopback address, this code is wrong.

Reply:
Good catch. I've updated the PR to actually pick up all the DNS configuration from passt, which was the right thing to do.